src/condor_tools/htcondor_cli/eventlog.py

```diff
@@ -0,0 +1,117 @@
+        log_file = Path(log_file)
+        if not log_file.exists():
+            raise FileNotFoundError(f"Could not find file: {str(log_file)}")
+        if os.access(log_file, os.R_OK) is False:
+            raise PermissionError(f"Could not access file: {str(log_file)}")
+        
+        # dictionary to store summary of events for each job
+        event_summaries = {}
+
+        with htcondor.JobEventLog(str(log_file)) as event_log:
+            for event in event_log.events(0):
+                job_id = str(event.cluster) + "." + str(event.proc)
+                event_summary = event_summaries.get(job_id, {})
+                if event.type == htcondor.JobEventType.SUBMIT:
+                    ip_address = re.search(r"\d+\.\d+\.\d+\.\d+", event.get("SubmitHost")).group()
+                    event_summaries[job_id] = {"Host": ip_address, "Evictions": 0, "Executed": False}
+                elif event.type == htcondor.JobEventType.JOB_ABORTED and event_summary.get("Executed", False) is False:
+                    # remove job from dictionary if it was aborted before being executed
+                    event_summaries.pop(job_id)
+                elif event.type == htcondor.JobEventType.EXECUTE:
+                    # get hostname if possible
+                    if event.get("SlotName") is not None:
+                        hostname = event.get("SlotName")
+                        event_summary["Host"] = hostname
+                    # get last execution time
+                    start_time = datetime.fromisoformat(event.get("EventTime")).strftime("%-m/%-d %H:%M")
+                    event_summary["Start Time"] = start_time
+                    event_summary["last_exec_time"] = start_time
+                    # set boolean that job has been executed
+                    event_summary["Executed"] = True
+                elif event.type == htcondor.JobEventType.JOB_EVICTED or event.type == htcondor.JobEventType.JOB_TERMINATED:
+                    # get good time and cpu usage
+                    cpu_usage_key = "RunRemoteUsage" if event.type == htcondor.JobEventType.JOB_EVICTED else "TotalRemoteUsage"
+                    cpu_usage = str(event.get(cpu_usage_key))
+                    user_time = re.search(r"Usr \d+ (\d+):(\d+):(\d+),", cpu_usage)
+                    sys_time = re.search(r"Sys \d+ (\d+):(\d+):(\d+)", cpu_usage)
+                    user_minutes, user_seconds = user_time.group(1, 2)
+                    sys_minutes, sys_seconds = sys_time.group(1, 2)
+                    total_minutes = int(user_minutes) + int(sys_minutes)
+                    total_seconds = int(user_seconds) + int(sys_seconds)
+                    cpu_usage_formatted = f"0+{total_minutes:02d}:{total_seconds:02d}"
+                    event_summary["CPU Usage"] = cpu_usage_formatted
+                    if event.type == htcondor.JobEventType.JOB_EVICTED:
+                        event_summary["Evictions"] = event_summary.get("Evictions", 0) + 1
```

Comment:
I like the defensive programming here (not assuming that we saw a `SUBMIT` event before we see a `JOB_EVICTED` event).